openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -426,7 +427,32 @@ public static String getBallerinaMediaType(String mediaType) {
             case MediaType.APPLICATION_OCTET_STREAM:
             case APPLICATION_PDF:
             case ANY_TYPE:
-                return SyntaxKind.BYTE_KEYWORD.stringValue() + SQUARE_BRACKETS;
+                return HTTP_RESPONSE;
+            default:
+                return SyntaxKind.JSON_KEYWORD.stringValue();
+            // TODO: fill other types
+        }
+    }
+
+    /**
+     * Generate BallerinaMediaType for all the input mediaTypes.
+     */
+    public static String getBallerinaMediaTypeForRequest(String mediaType) {
+
+        switch (mediaType) {
+            case MediaType.APPLICATION_JSON:
+                return SyntaxKind.JSON_KEYWORD.stringValue();
+            case MediaType.APPLICATION_XML:
+                return SyntaxKind.XML_KEYWORD.stringValue();
+            case MediaType.APPLICATION_FORM_URLENCODED:
+            case MediaType.TEXT_HTML:
+            case MediaType.TEXT_PLAIN:
+                return STRING_KEYWORD.stringValue();
+            case IMAGE_PNG:
+            case MediaType.APPLICATION_OCTET_STREAM:
+            case APPLICATION_PDF:
+            case ANY_TYPE:
+                return HTTP_REQUEST;
             default:
                 return SyntaxKind.JSON_KEYWORD.stringValue();
```

Comment:
Can't we map all the unhandled payload content types to `http:Request` by setting the default behaviour as `http:Request` instead of `json` ?

Reply:
Was there a special requirement to make the default is `json` ?

Reply:
@NipunaRanasinghe

Reply:
Normally we used `http:Request` where we can not handle and predict the request payload type. Therefore making `http:Request` default in the switch case makes sense for all the wild card behaviours as well :) WDYT?

Reply:
Added a new issue to address this comment. https://github.com/ballerina-platform/openapi-tools/issues/636